MSI/Day12/실패율.js

```diff
@@ -0,0 +1,13 @@
+function solution(N, stages) {
+    const fail = [];
+    let num2 = stages.length;
+
+    for (let i = 1; i <= N; i++) {
+        const num = stages.filter(v => v === i).length;
+        fail.push([i, num / num2]);
+        num2 -= num;
+    }
+
+    const answer = fail.sort((a, b) => b[1] - a[1]);
+    return answer.map(v => v[0]);
+}
```

Comment:
```suggestion
    let stateLength= stages.length;

    for (let i = 1; i <= N; i++) {
        const currentStageCount = stages.filter(stage => stage === i).length;
        fail.push([i, currentStageCount/ stateLength]);
        stateLength -= currentStageCount;
    }

    const answer = fail.sort((a, b) => b[1] - a[1]);
    return answer.map(v => v[0]);
}
```
`num1`,`num2`같은 알아보기 힘든 네이밍보다는 확실한 의미를 갖는 변수 네이밍이 좋습니다!

Reply:
수정했습니다 감사합니다!